python/hsfs/feature_group.py

```diff
@@ -705,6 +709,7 @@ def save(
 
         # fg_job is used only if the hive engine is used
         fg_job = self._feature_group_engine.save(self, feature_dataframe, write_options)
+        self._code_engine.compute_code(self)
```

Comment:
You need to add these calls also on the `OnDemandFeatureGroup` class - for the save and insert method.

Reply:
OnDemandFeatureGroup doesn't have insert, only save.